src/attributes/classes.js

```diff
@@ -158,12 +156,13 @@ jQuery.fn.extend({
 	},
 
 	hasClass: function( selector ) {
-		var className = " " + selector + " ",
-			i = 0,
-			l = this.length;
-		for ( ; i < l; i++ ) {
-			if ( this[i].nodeType === 1 &&
-				( " " + getClass( this[i] ) + " " ).replace( rclass, " " )
+		var className, elem,
+			i = 0;
+
+		className = " " + selector + " ";
```

Comment:
Since we don't require to use single var per scope, we could do here `var className = " " + selector + " ";`, i bet you wouldn't became smaller, but i guess more readable